includes/object-cache.php

```diff
@@ -514,11 +514,18 @@ public function __construct( $fail_gracefully = true ) {
             }
 
             if ( defined( 'WP_REDIS_CLUSTER' ) ) {
-                $connectionID = current( $this->build_cluster_connection_array() );
 
-                $this->diagnostics[ 'ping' ] = ($client === 'predis')
-                    ? $this->redis->getClientFor( $connectionID )->ping()
-                    : $this->redis->ping( $connectionID );
+	            if ( is_string( WP_REDIS_CLUSTER ) ) {
+		            $connectionID = WP_REDIS_CLUSTER;
+	            } else {
+		            $connectionID = current( $this->build_cluster_connection_array() );
+	            }
+
+	            if ( $client === 'predis' ) {
+		            $this->diagnostics[ 'ping' ] = $this->redis->getClientFor( $connectionID )->ping();
+	            } elseif ( $client === 'phpredis' ) {
```

Comment:
This condition break Credit/Relay compatibility, just use an `else`, or keep the old format but fix the `$connectionID`.

Reply:
I made changes, but if constant WP_REDIS_CLUSTER is defined Exception will thrown here https://github.com/rhubarbgroup/redis-cache/blob/develop/includes/object-cache.php#L726